src/game_battlealgorithm.cpp

```diff
@@ -462,6 +466,7 @@ void Game_BattleAlgorithm::AlgorithmBase::GetResultMessages(std::vector<std::str
 
 		// If enemy is killed, it ends here
 		if (killed_by_attack_damage) {
+			out_replace.push_back(1);
```

Comment:
This out_replace mechanism looks really brittle. Can't we just edit the vector directly?